src/beacons/BeaconsContext.tsx

```diff
@@ -153,10 +154,12 @@ const BeaconsContextProvider: React.FC = ({children}) => {
     }
   }, [isBeaconsSupported]);
 
-  const deleteCollectedData = useCallback(async () => {
+  const deleteCollectedData = useCallback(() => {
     if (!isBeaconsSupported) return;
     if (isInitializedRef.current) {
-      await Kettle.deleteCollectedData();
+      Kettle.deleteCollectedData().catch((error) => {
+        Bugsnag.notify(error);
+      });
```

Comment:
If you have revoked permissions, will the initialised ref be false and data not deleted? Or is data deleted when you revoke?

Reply:
I think data is not removed when revoke collect data, only `delete collected data` removes it, as far I know.

Reply:
Good catch, I think it would actually require to init the sdk to delete the data in that case

Reply:
So maybe it is needed to init, delete, then stop again when `isInitializedRef.current` is false?

Reply:
But call revoke do not turn false: isInitializedRef.current, once is initialized the SDK, it remains initialized. Which is different from running.

Reply:
So in the case is false is because is either not onboarded or disabled in the remote config, for those cases in the debug menu beacons is not show and there is PR to do not show the privacy beacons related stuff if not onboarded.

Reply:
@jorelosorio Yes I confused stop with un-init, which is not a method and so won't happen until the user restarts the app.
So the beacons privacy will be hidden for the false case after another upcoming PR? I guess then the problem is avoided :+1:

Reply:
I'm not quite following what problem is avoided 😅  We still need to be able to delete when uninitialised? 🤔 

Since it seems like initialisation only happen when user are sharing:
```ts
if (permissions.length > 0) {
  await NativeModules.KettleSDKExtension.initializeKettleSDK();
  isInitializedRef.current = true;
}
```

Reply:
I was thinking if the button is hidden, but realising now that the Delete My Profile button should be there regardless of that flag. In which case, problem not avoided